stellargraph/layer/knowledge_graph.py

```diff
@@ -191,3 +192,135 @@ def build(self):
         x_out = self(x_inp)
 
         return x_inp, x_out
+
+
+class DistMultScore(Layer):
+    """
+    DistMult scoring Keras layer.
+
+    Original Paper: Embedding Entities and Relations for Learning and Inference in Knowledge
+    Bases. Bishan Yang, Wen-tau Yih, Xiaodong He, Jianfeng Gao, Li Deng. ICLR 2015
+
+    This combines subject, relation and object embeddings into a score of the likelihood of the
+    link.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def build(self, input_shape):
+        self.built = True
+
+    def call(self, inputs):
```

Comment:
And documentation about the input shapes here as well.

Side note, very minor typo in `ComplExScore.call` docstring that I missed before; the input shapes are `(batch size, 1, embedding dim)` not `(batch size, embedding dim)`

```suggestion
    def call(self, inputs):
        """
        Applies the layer.

        Args:

            inputs: a list of 3 tensors (each batch size, 1, embedding dimension), where the three
                consecutive tensors are the embeddings of the the subject, relation and
                object embeddings, respectively.
        """
```